File: android/src/main/java/com/salkuadrat/music/Music.java
package com.salkuadrat.music;

import android.graphics.Bitmap;
import android.support.v4.media.session.MediaSessionCompat;

public class Music {
    String id = "";
    String title = "";
    String artist = "";
    String album = "";
    Bitmap image;
    String imageUrl = "";
    int duration = 0;
    int position = 0;
    boolean showPrevious = false;
    boolean showNext = false;
    boolean isPlaying = false;
    boolean isLoading = false;
    MediaSessionCompat session;
}

File: lib/src/music.dart
/// Class to create instance of music to play using MusicPlayer
class Music {
  /// Music ID
  final String id;

  /// Music File URL
  final String url;

  /// Music Singer
  final String artist;

  /// Music Album Name
  final String? album;

  /// Music Title
  final String title;

  /// Music Cover URL
  final String image;

  /// Music Duration
  final Duration? duration;

  Music({
    String? id,
    required this.url,
    required this.artist,
    this.album,
    required this.title,
    required this.image,
    this.duration,
  }) : this.id = id ?? url;
}
